### PR TITLE
[TTAHUB-1327] Add unique ids to buttons across the applications for analytics

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Following the request in the Jira, unique ids were added to assorted buttons on assorted pages. This is for the easy attachment of handlers in Google Tag Manager.

## How to test
Compare the rendered DOM with the list in the Jira ticket. Note the comments where I've explained changes I made to the original request

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1327

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js [hunk 1]
```diff
@@ -136,7 +136,7 @@ const Draft = ({
         <div className="margin-top-3">
           <ApproverStatusList approverStatus={approverStatusList} />
         </div>
-        <Button disabled={!connectionActive} type="submit">Submit for approval</Button>
+        <Button disabled={!connectionActive} id="draft-review-submit" type="submit">Submit for approval</Button>
```
Comment (GarrettEHill) [inline frontend/src/pages/ActivityReport/Pages/Review/Submitter/Draft.js:139]: Does this need a unique id?

File: frontend/src/pages/RecipientRecord/pages/PrintGoals.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ export default function PrintGoals({ location, recipientId, regionId }) {
 
   return (
     <div className="margin-top-2 margin-left-2 ttahub-print-goals">
-      <PrintToPdf />
+      <PrintToPdf id="print-goals" />
```
Comment (GarrettEHill) [inline frontend/src/pages/RecipientRecord/pages/PrintGoals.js:70]: Does this need a unique id?

File: frontend/src/widgets/TotalHrsAndRecipientGraph.js [hunk 3]
```diff
@@ -229,13 +229,14 @@ export function TotalHrsAndRecipientGraph({ data, loading }) {
         <Grid row className="position-relative margin-bottom-2">
           <Grid desktop={{ col: 'auto' }} mobileLg={{ col: 8 }}><h2 className="ttahub--dashboard-widget-heading margin-0">Total TTA hours</h2></Grid>
           <Grid desktop={{ col: 'auto' }} className="ttahub--show-accessible-data-button desktop:margin-y-0 mobile-lg:margin-y-1">
-            { !showAccessibleData && <MediaCaptureButton className="margin-x-2" reference={widget} buttonText="Save screenshot" /> }
+            { !showAccessibleData && <MediaCaptureButton id="rd-save-screenshot" className="margin-x-2" reference={widget} buttonText="Save screenshot" /> }
```
Comment (GarrettEHill) [inline frontend/src/widgets/TotalHrsAndRecipientGraph.js:232]: Does this need a unique id?

**Review by GarrettEHill — APPROVED** [review]

**Comment by GarrettEHill** [comment]
do we also want to add this type of id to anchor tags too?

File: frontend/src/pages/ApprovedActivityReport/index.js [hunk 4]
```diff
@@ -223,9 +223,10 @@ export default function ApprovedActivityReport({ match, user }) {
         : null}
       <Grid row>
         {navigator && navigator.clipboard
-          ? <button type="button" className="usa-button no-print" disabled={modalRef && modalRef.current ? modalRef.current.modalIsOpen : false} onClick={handleCopyUrl}>Copy URL Link</button>
+          ? <button id="approved-URL" type="button" className="usa-button no-print" disabled={modalRef && modalRef.current ? modalRef.current.modalIsOpen : false} onClick={handleCopyUrl}>Copy URL Link</button>
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/ApprovedActivityReport/index.js:226]: Not to be picky but should 'URL' be in caps?

Comment (AdamAdHocTeam) [reply frontend/src/pages/ApprovedActivityReport/index.js:226]: From JIRA: Use lower case letters and dashes between words. For example, id="example-button"

File: frontend/src/pages/RecipientRecord/pages/TTAHistory.js [hunk 5]
```diff
@@ -115,6 +115,7 @@ export default function TTAHistory({
             filters={filtersToApply}
             showFilter={false}
             tableCaption="Approved activity reports"
+            exportIdPrefix="ttahistory-"
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/RecipientRecord/pages/TTAHistory.js:118]: Another small thing tta-history

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good added some comments and suggestions below to add:

**Missing Id:**
- new activity report button on ar landing
- Please leave feedback (survey) on ar landing
- Add new objective on ar goal creation
- Select and upload file attachments on ar creation
- Add next step on ar creation.
- Submit button for an approver submitting their approval

**Pull request merged**